azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java

```diff
@@ -257,10 +255,17 @@ private void checkAllTriggers() throws TriggerManagerException {
           scannerStage = "Checking for trigger " + t.getTriggerId();
 
           if (t.getStatus().equals(TriggerStatus.READY)) {
-            if (t.triggerConditionMet()) {
+
+            /**
+             * Prior to this change, expiration condition should never be called though
+             * we have some related code here. ExpireCondition used the same BasicTimeChecker
+             * as triggerCondition do. As a consequence, we need to figure out a way to distinguish
+             * the previous ExpireCondition and this commit's ExpireCondition.
+             */
+            if (t.getExpireCondition().getExpression().contains("EndTimeChecker") && t.expireConditionMet()) {
```

Comment:
I know this is hacky. This hack is intended for distinguishing existing schedules (created before the release of this PR). In other words, I don't need this hack if AZ doesn't have any schedule today.
Becuase existing triggers have the triggerCondition same as expireCondition, we need to tell the difference by this thack.